### DISCUSSION TITLE
Attempting to recover zpools created on version 0.6.3-1.2 on Centos7.2

### DISCUSSION THREAD

**Discussion by scantwell** [openzfs/zfs#12861] [Q&A] [answered]
My OS drive died on my Centos7.2 box that created the zpools. I've since bought a new drive and install Ubuntu. After installing zfs the zpool import command returns no available pools. Is there a way I can recover them?

**Comment by rincebrain** [comment]
On what disks were the pools created? Was LUKS or anything involved? What do the partition tables on the drives in question look like? 

Normally `zpool import` would just show them, so there's presumably something slightly different about their setup so it's not showing them (and since `zpool import` still lists my circa-2010 pool as expected, I don't think it's to do with being too old. :))

You could also examine `zpool import -D`, but that should only help if your pools were zpool destroyed...

**Reply by scantwell** [reply]
HI rincebrain,

Thank you for the response. I have two WD red 2TB drives in a mirror. I am unsure if I had LUKS enabled on them, I don't believe so. There were no partitions on the disks the pool "tank" was given both disks. Tank contained two subvolumes media and home as shown below in the fstab file. 

fstab
UUID=fbc35508-2a12-4234-8246-f626d7cbdf2f /tank btrfs defaults,compress=lzo,device=/dev/sdb,device=/dev/sdc 0 0
UUID=fbc35508-2a12-4234-8246-f626d7cbdf2f /home btrfs defaults,compress=lzo,subvol=home,device=/dev/sdb,device=/dev/sdc 0 0
UUID=fbc35508-2a12-4234-8246-f626d7cbdf2f /var/lib/plexmediaserver/Library/media btrfs defaults,compress=lzo,subvol=media,device=/dev/sdb,device=/dev/sdc 0 0

Would a new version of the library on a different operating system (Ubuntu) be able to read an older version like 0.6.3-1.2? I was using dkms on Centos7.2 and I'm not sure if that is what Ubuntu is using. IOW, does installing zfs with dkms vs kmod/kabi make a difference on being able to see the pools?

sudo zpool import -D
no pools available to import

sudo zdb -l /dev/sdc
failed to unpack label 0
failed to unpack label 1
failed to unpack label 2
failed to unpack label 3

I do see zfs data including "trunk" in a hexdump of the disk so I know it hasn't been erased for some odd reason.

**Reply by rincebrain** [reply]
Those aren't ZFS mount lines, those are btrfs mount lines, which would go some way toward explaining why ZFS isn't finding useful data right now. I would suggest you explore importing the btrfs filesystems instead unless you know something I don't about this configuration.

No, kmod versus DKMS should make no functional difference once the modules are loaded, and yes, newer versions should be perfectly happy to read older ones.

**Comment by scantwell** [comment, marked as answer]
User error. This ticket can be closed
